driver/ni/scale.h

```diff
@@ -0,0 +1,287 @@
+struct LinearScale {
+    float64 slope;
+    float64 offset;
+
+    LinearScale() = default;
+
+
+    LinearScale(config::Parser &parser)
+        : slope(parser.required<double>("slope")),
+          offset(parser.required<double>("y_intercept")) {
+        if (!parser.ok()) LOG(ERROR) <<
+                          "[ni.analog] failed to parse custom linear configuration";
+    }
+};
+
+///////////////////////////////////////////////////////////////////////////////////
+//                                    MapScale                                   //
+///////////////////////////////////////////////////////////////////////////////////
+struct MapScale {
+    float64 prescaled_min;
+    float64 prescaled_max;
+    float64 scaled_min;
+    float64 scaled_max;
+
+    MapScale() = default;
+
+    MapScale(config::Parser &parser)
+        : prescaled_min(parser.required<double>("pre_scaled_min")),
+          prescaled_max(parser.required<double>("pre_scaled_max")),
+          scaled_min(parser.required<double>("scaled_min")),
+          scaled_max(parser.required<double>("scaled_max")) {
+        if (!parser.ok()) LOG(ERROR) <<
+                          "[ni.analog] failed to parse custom map configuration";
+    }
+};
+
+///////////////////////////////////////////////////////////////////////////////////
+//                                  PolynomialScale                              //
+///////////////////////////////////////////////////////////////////////////////////
+struct PolynomialScale {
+    std::vector<double> forward_coeffs;
+    std::vector<double> reverse_coeffs;
+    uint32_t num_coeffs;
+    float64 min_x;
+    float64 max_x;
+    int32 num_points;
+    int32 poly_order;
+
+    PolynomialScale() = default;
+
+    PolynomialScale(config::Parser &parser)
+        : num_coeffs(parser.required<int>("num_coeffs")),
+          min_x(parser.required<double>("min_x")),
+          max_x(parser.required<double>("max_x")),
+          num_points(parser.optional<int>("num_reverse_coeffs", 0)),
+          poly_order(parser.required<int>("poly_order")) {
+            
+        forward_coeffs.resize(num_coeffs*2);
+        reverse_coeffs.resize(num_coeffs*2);
+
+        if (!parser.ok()) {
+            LOG(ERROR) << "[ni.analog] failed to parse custom polynomial scale configuration";
+            return;
+        }
+
+        //TODO: handle if there is reverse coeffs of different size than forward coeffs
+
+        //get json from parser
+        json j = parser.get_json();
+        // get forward coeffs (prescale -> scale conversions)
+        if (!j.contains("coeffs")) {
+            LOG(ERROR) << "[ni.analog] failed to parse custom polynomial scale configuration: missing coeffs";
+            return; 
+        }
+
+        for (int i = 0; i < num_coeffs; i++) forward_coeffs[i] = j["coeffs"][i];
+        
+        // get reverse coeffs (scale -> prescale conversions)
+        // TODO: reverse coeffs might be smaller than forward_coeffs
```

Comment:
Address `TODO` and `FIXME`

Reply:
will not address in this pr. will address for release which provides polynomial support